modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTaskRunner.java

```diff
@@ -0,0 +1,166 @@
+ * and synchronous versus asynchronous.
+ * <p>
+ * A calculation runner is typically obtained using the static methods on this interface.
+ * The instance contains an executor thread-pool, thus care should be taken to ensure
+ * the thread-pool is correctly managed. For example, try-with-resources could be used:
+ * <pre>
+ *  try (CalculationTaskRunner runner = CalculationTaskRunner.ofMultiThreaded()) {
+ *    // use the runner
+ *  }
+ * </pre>
+ */
+public interface CalculationTaskRunner extends AutoCloseable {
+
+  /**
+   * Creates a standard multi-threaded calculation task runner capable of performing calculations.
+   * <p>
+   * This factory creates an executor basing the number of threads on the number of available processors.
+   * It is recommended to use try-with-resources to manage the runner:
+   * <pre>
+   *  try (CalculationTaskRunner runner = CalculationTaskRunner.ofMultiThreaded()) {
+   *    // use the runner
+   *  }
+   * </pre>
+   * 
+   * @return the calculation task runner
+   */
+  public static CalculationTaskRunner ofMultiThreaded() {
+    return DefaultCalculationTaskRunner.ofMultiThreaded();
+  }
+
+  /**
+   * Creates a calculation task runner capable of performing calculations, specifying the executor.
+   * <p>
+   * It is the callers responsibility to manage the life-cycle of the executor.
+   * 
+   * @param executor  the executor to use
+   * @return the calculation task runner
+   */
+  public static CalculationTaskRunner of(ExecutorService executor) {
+    return DefaultCalculationTaskRunner.of(executor);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates the tasks that perform the individual calculations.
+   * <p>
+   * This returns an instance of that contains one {@link CalculationTask} for each combination
+   * of target and column, that can be visualized as a grid of columns with a row for each target.
+   * <p>
+   * The tasks object provides access to the list of targets, columns and tasks.
+   * It also provides the ability to query what market data is needed to perform pricing.
+   * <p>
+   * Obtaining an instance of {@code CalculationTasks} for a set of trades and measures
+   * performs some one-off initialization. Providing access to the instance allows the initialization
+   * to occur once, which could be a performance optimization if many different calculations
+   * are performed with the same set of trades and measures.
+   *
+   * @param targets  the targets for which values of the measures will be calculated
+   * @param columns  the configuration for the columns that will be calculated, including the measure and
+   *   any column-specific overrides
+   * @param calculationRules  the rules defining how the calculation is performed
+   * @return the tasks that perform the calculations
+   */
+  public abstract CalculationTasks createTasks(
```

Comment:
I don't think this method adds anything over just calling `CalculationTasks.of(targets, columns, calculationRules)`.

We put an equivalent method on the old `CalculationRunner` just in case we ever needed some state from the runner but we never did in all the time it was there. So I think it's safer (and simpler) to get rid of it.